test/docfx.Test/lib/HtmlUtilityTest.cs

```diff
@@ -163,4 +163,46 @@ public static void Encode(string input, string expected)
     {
         Assert.Equal(expected, HtmlUtility.Encode(input));
     }
+
+    [Theory]
+    [InlineData("<div style='a'></div>", "{\"div\":{\"style\":1}}")]
+    [InlineData("<div><script></script></div>", "{\"div\": {\"\": 1},\"script\": {\"\": 1}}")]
+    public static void CollectHtmlUsage(string input, string expected)
+    {
+        var expectedCount = JsonUtility.DeserializeData<Dictionary<string, Dictionary<string, int>>>(expected, null);
+        var elementCount = new Dictionary<string, Dictionary<string, int>>();
+        HtmlUtility.CollectHtmlUsage(input, elementCount);
+        Assert.True(CompareDictionaries(expectedCount, elementCount));
```

Comment:
Instead of write a custom compare dictionary, simply compare the serialized JSON.
```suggestion
        Assert.Equal(expected, JsonSerializer.Serialize(elementCount));
```

Reply:
Does JsonSerializer.Serialize sort the key-value pairs by key?

Reply:
It doesn't sort, but the result should be deterministic. Most likely it'll use dictionary enumeration order which is dictionary insertion order which is deterministic per input HTML.

Reply:
.NET has randomized string hashing but Dictionary\<TKey, TValue\> disables that until it gets more than 100 hash collisions during the same insertion, so it is still deterministic with two elements.

- <https://github.com/dotnet/runtime/blob/v6.0.0/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs#L62-L72>
- <https://github.com/dotnet/runtime/blob/v6.0.0/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs#L676-L682>